README.md

```diff
@@ -0,0 +1,496 @@
+    * [Miscellaneous](#miscellaneous)
+
+<a name="code_layout"></a>
+## Code layout
+
+<a name="tabs_or_spaces"></a>
+### Tabs or Spaces?
+
+Use **spaces only**, with **2 spaces** per indentation level. Never mix tabs and spaces.
+
+<a name="maximum_line_length"></a>
+### Maximum Line Length
+
+Limit all lines to a maximum of 79 characters.
+
+<a name="blank_lines"></a>
+### Blank Lines
+
+Separate top-level function and class definitions with a single blank line.
+
+Separate method definitions inside of a class with a single blank line.
+
+Use a single blank line within the bodies of methods or functions in cases where this improves readability (e.g., for the purpose of delineating logical sections).
+
+<a name="trailing_whitespace"></a>
+### Trailing Whitespace
+
+Do not include trailing whitespace on any lines.
+
+<a name="optional_commas"></a>
+### Optional Commas
+
+Avoid the use of commas before newlines when properties or elements of an Object or Array are listed on separate lines.
+
+```coffeescript
+# Yes
+foo = [
+  'some'
+  'string'
+  'values'
+]
+bar:
+  label: 'test'
+  value: 87
+
+# No
+foo = [
+  'some',
+  'string',
+  'values'
+]
+bar:
+  label: 'test',
+  value: 87
+```
+
+<a name="encoding"></a>
+### Encoding
+
+UTF-8 is the preferred source file encoding.
+
+<a name="module_imports"></a>
+## Module Imports
+
+If using a module system (CommonJS Modules, AMD, etc.), `require` statements should be placed on separate lines.
+
+```coffeescript
+require 'lib/setup'
+Backbone = require 'backbone'
+```
+These statements should be grouped in the following order:
+
+1. Standard library imports _(if a standard library exists)_
+2. Third party library imports
+3. Local imports _(imports specific to this application or library)_
+
+<a name="whitespace"></a>
+## Whitespace in Expressions and Statements
+
+Avoid extraneous whitespace in the following situations:
+
+- Immediately inside parentheses, brackets or braces
+
+    ```coffeescript
+       ($ 'body') # Yes
+       ( $ 'body' ) # No
+    ```
+
+- Immediately before a comma
+
+    ```coffeescript
+       console.log x, y # Yes
+       console.log x , y # No
+    ```
+
+Additional recommendations:
+
+- Always surround these binary operators with a **single space** on either side
+
+    - assignment: `=`
+
+        - _Note that this also applies when indicating default parameter value(s) in a function declaration_
+
+           ```coffeescript
+           test: (param = null) -> # Yes
+           test: (param=null) -> # No
+           ```
+
+    - augmented assignment: `+=`, `-=`, etc.
+    - comparisons: `==`, `<`, `>`, `<=`, `>=`, `unless`, etc.
+    - arithmetic operators: `+`, `-`, `*`, `/`, etc.
+
+    - _(Do not use more than one space around these operators)_
+
+        ```coffeescript
+           # Yes
+           x = 1
+           y = 1
+           fooBar = 3
+
+           # No
+           x      = 1
+           y      = 1
+           fooBar = 3
+        ```
+
```

Comment:
In this section, do we want to include the spacing rules from the JS style guide:
* [Arrays](https://github.com/distil/js-style-guide#whitespace--in-brackets)
* [Objects (with explicit curly braces)](https://github.com/distil/js-style-guide#whitespace--in-braces)